pyccl/tracers.py

```diff
@@ -185,6 +185,79 @@ def __init__(self):
         # Do nothing, just initialize list of tracers
         self._trc = []
 
+    def __eq__(self, other):
+        # Check the object class.
+        if self.__class__ is not other.__class__:
+            return False
+
+        # If the tracer collections are empty, return early.
+        if not (self or other):
+            return True
+
+        # If the tracer collections are not the same length, return early.
+        if len(self._trc) != len(other._trc):
+            return False
+
+        # Check `der_angles` & `der_bessel` for each tracer in the collection.
+        bessel = self.get_bessel_derivative(), other.get_bessel_derivative()
+        angles = self.get_angles_derivative(), other.get_angles_derivative()
+        if not (np.array_equal(*bessel) and np.array_equal(*angles)):
+            return False
+
+        # Check the kernels.
+        kwargs = {"atol": 0, "rtol": 1e-12, "equal_nan": True}
+        for t1, t2 in zip(self._trc, other._trc):
+            if bool(t1.kernel) ^ bool(t2.kernel):
+                # only one of them has a kernel
+                return False
+            if t1.kernel is None:
+                # none of them has a kernel
+                continue
+            if not np.allclose(_get_spline1d_arrays(t1.kernel.spline),
+                               _get_spline1d_arrays(t2.kernel.spline),
+                               **kwargs):
```

Comment:
No, please. Check that they are exactly the same

Reply:
done